GVFS/GVFS.Platform.Mac/MacFileSystem.cs

```diff
@@ -29,6 +30,24 @@ public override bool IsSocket(string fileName)
             return NativeStat.IsSock(statBuffer.Mode);
         }
 
+        public override bool IsFileSystemSupported(string path, out string error)
+        {
+            string lowerCaseFile = Guid.NewGuid().ToString().ToLower();
+            string upperCaseFile = lowerCaseFile.ToUpper();
+
+            File.Create(Path.Combine(path, lowerCaseFile));
+            if (File.Exists(upperCaseFile))
+            {
+                File.Delete(lowerCaseFile);
+                error = "Case sensitive unsupported";

```

Comment:
Minor: Can you make this error message a bit more descriptive?  Maybe something like "VFS for Git does not support case sensitive filesystems"